test/parallel/test-inspector-async-hook-after-done.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const common = require('../common');
+
+common.skipIfInspectorDisabled();
+
+const { Worker } = require('worker_threads');
+const { Session } = require('inspector');
+
+const session = new Session();
+
+let done = false;
+
+session.connect();
+
+session.on('NodeWorker.attachedToWorker', ({ params: { sessionId } }) => {
+  let id = 1;
+  function postToWorker(method, params) {
+    session.post('NodeWorker.sendMessageToWorker', {
+      sessionId,
+      message: JSON.stringify({ id: id++, method, params })
+    }, () => console.log(`Message ${method} received the response`));
+  }
+
+  // Wait for the notification
+  function onMessageReceived({ params: { message } }) {
+    if (!message ||
+      JSON.parse(message).method !== 'NodeRuntime.waitingForDisconnect') {
+      session.once('NodeWorker.receivedMessageFromWorker', onMessageReceived);
+      return;
+    }
+    // Force a call to node::inspector::Agent::ToggleAsyncHook by changing the
+    // async call stack depth
+    postToWorker('Debugger.setAsyncCallStackDepth', { maxDepth: 1 });
+    // This is were the original crash happened
+    session.post('NodeWorker.detach', { sessionId }, () => {
+      done = true;
+    });
+  }
+
+  onMessageReceived({ params: { message: null } });
+  // Enable the debugger, otherwise setAsyncCallStackDepth does nothing
+  postToWorker('Debugger.enable');
+  // Start waiting for disconnect notification
+  postToWorker('NodeRuntime.notifyWhenWaitingForDisconnect', { enabled: true });
+  // start worker
+  postToWorker('Runtime.runIfWaitingForDebugger');
+});
+
+session.post('NodeWorker.enable', { waitForDebuggerOnStart: true }, () => {
+  new Worker('console.log("Worker is done")', { eval: true });
+  // Inspector API does not hold any async handles so the test may terminate
+  // before the test code runs to a completion.
+  (function waitForDone() {
+    if (!done) setTimeout(waitForDone, 0);
+  })();
```

Comment:
Can we avoid this by e.g. waiting for the Worker’s `exit` event?

Reply:
Done.